botocore/client.py

```diff
@@ -178,6 +178,15 @@ def _create_s3_redirector(self, client, endpoint_bridge):
             return
         S3RegionRedirector(endpoint_bridge, client).register()
 
+    def _get_default_s3_region(self, service_name, endpoint_bridge):
+        # If a user is providing a custom URL, the endpoint resolver will
+        # refuse to infer a signing region. If we want to default to s3v4,
```

Comment:
Why is this?  Is this intentional?  What's the point of not inferring a signing name if you provide an endpoint_url, but inferring a region if you don't provide an endpoint_url?